lib/encoder.js

```diff
@@ -535,6 +535,19 @@ function JPEGEncoder(quality) {
 				writeByte(std_ac_chrominance_values[p]);
 			}
 		}
+		
+		function writeCOM(comments)
+		{
+			if (typeof comments === "undefined") return;
```

Comment:
```suggestion
			if (typeof comments !== 'string') return;
```

perhaps?

Reply:
This will not work, I am afraid. The type is 'object' as it is an array.

Reply:
Ah sorry, misunderstood your comment. I will change that tomorrow.